### PR TITLE
Add Dockerfile and entrypoint script

### PR DESCRIPTION
**Pull request by agalue**
To be able to use the gRPC server with Docker or Kubernetes, I've added a simple Dockerfile with two phases, one to compile the JAR (to avoid requiring having a JDK installed), and another to build the image with the compiled JAR and the entry point script.

The entry point script exposes all possible variables that can be passed to the gRPC server via environment variables.

The generated image is how I've been testing the gRPC in Docker and Kubernetes so far.

### CODE REVIEW

**Review by cgorantla — APPROVED** [review]

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  env_var=$(echo "$VAR" | cut -d= -f1)
+  if [[ $env_var =~ ^PRODUCER_ || $env_var =~ ^CONSUMER_ ]]; then
+    key="org.opennms.core.ipc.grpc.kafka."$(echo "$env_var" | tr '[:upper:]' '[:lower:]' | tr _ .)
+    val=${!env_var}
+    echo "[Configuring] '$key'='$val'"
+    OPTIONS+=("-D$key=$val")
+  fi
+done
+
+export JAVA_TOOL_OPTIONS="${JAVA_OPTS} $(join ' ' ${OPTIONS[@]})"
```
Comment (indigo423) [inline docker-entrypoint.sh:37]: Following a more common practice for a jvm.options file like Cassandra or Elasticsearch does would be nice, but not super critical :)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+WORKDIR /app
+ENV GRPC_HEALTH_PROBE_VERSION=v0.3.2
+RUN wget -qOgrpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-amd64 && \
+    chmod +x grpc_health_probe && \
+    git clone https://github.com/OpenNMS/grpc-server.git && \
+    cd grpc-server && \
+    git checkout -b feature/health-check origin/feature/health-check && \
+    mvn package
+
+FROM openjdk:8-jdk-slim
```
Comment (indigo423) [inline Dockerfile]: does it not work with a more modern JDK?

**Review by indigo423 — APPROVED** [review]
Nothing critical to me, just the comments. LGTM 👍

**Comment by agalue** [comment]
I've updated `pom.xml` and `Dockerfile` to fully support OpenJDK 11, to compile and run the gRPC server.

I understand that this change, even if it is minimal, might be out of the scope, but I decided to include it. If you think we need a separate PR for it, please let me know.

I tested it, and everything still works as expected.

File: Dockerfile (hunk 2 above)
Comment (agalue) [reply Dockerfile]: Added support for OpenJDK 11

File: docker-entrypoint.sh (hunk 1 above)
Comment (agalue) [reply docker-entrypoint.sh:37]: `JAVA_TOOL_OPTIONS` was the only way I was able to make the solution work, as putting all the required flags in `JAVA_OPTS` never worked for me, at least with OpenJDK 8. Also, `JAVA_TOOL_OPTIONS` has the benefit of showing all the flags via `STDOUT` when the program starts.

**Pull request merged**